extensions/csrc/cuda/context_kv_cache_memcpy_kernel.cu

```diff
@@ -93,76 +95,61 @@ void apply_context_kv_cache_memcpy(
-                key.data_ptr<scalar_t>(),
-                value.data_ptr<scalar_t>(),
-                key_cache.data_ptr<scalar_t>(),
-                value_cache.data_ptr<scalar_t>(),
-                sequence_lengths.data_ptr<int>(),
-                cu_seqlens.data_ptr<int>(),
-                block_tables.data_ptr<int>(),
-                head_num,
-                head_dim,
-                block_size,
-                batch_size,
-                block_table_stride,
-                key_stride,
-                value_stride
-            );
-            break;
-        default:
-            AT_ERROR("Unsupported vectorized size ", vec_size);
-            break;
+#define CONTEXT_KV_CACHE_MEMCOPY_KERNEL_LAUNCH(__aligned, __vec_size)                                   \
+    do {                                                                                                \
+        context_kv_cache_memcpy_kernel<scalar_t, __aligned, __vec_size><<<grid, block, 0, stream>>>(    \
+                key.data_ptr<scalar_t>(),                                                               \
+                value.data_ptr<scalar_t>(),                                                             \
+                key_cache.data_ptr<scalar_t>(),                                                         \
+                value_cache.data_ptr<scalar_t>(),                                                       \
+                sequence_lengths.data_ptr<int>(),                                                       \
+                cu_seqlens.data_ptr<int>(),                                                             \
+                block_tables.data_ptr<int>(),                                                           \
+                head_num,                                                                               \
+                head_dim,                                                                               \
+                block_size,                                                                             \
+                batch_size,                                                                             \
+                block_table_stride,                                                                     \
+                key_stride,                                                                             \
+                value_stride                                                                            \
+            );                                                                                          \
+    } while(0)
+
+#define CONTEXT_KV_CACHE_MEMCOPY_KERNEL_LAUNCH_VEC_SIZE_CASE(__aligned)                                 \
+    do {                                                                                                \
+        switch (vec_size) {                                                                             \
```

Comment:
Where does `vec_size` come from

Reply:
![image](https://github.com/hpcaitech/ColossalAI/assets/26091437/6421668b-52cc-45d2-aad1-53e96409cc2e)
The vec_size is defined here.

Reply:
Got it.